google/auth/transport/aiohttp_req.py

```diff
@@ -0,0 +1,297 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Transport adapter for Async HTTP (aiohttp)."""
+
+from __future__ import absolute_import
+
+import asyncio
+import functools
+import logging
+
+
+import aiohttp
+import six
+
+
+from google.auth import exceptions
+from google.auth import transport
+from google.auth.transport import requests
+
+
+_OAUTH_SCOPES = [
+    "https://www.googleapis.com/auth/appengine.apis",
+    "https://www.googleapis.com/auth/userinfo.email",
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+# Timeout can be re-defined depending on async requirement. Currently made 60s more than
+# sync timeout.
+_DEFAULT_TIMEOUT = 180  # in seconds
+
+
+class _Response(transport.Response):
+    """
+    Requests transport response adapter.
+
+    Args:
+        response (requests.Response): The raw Requests response.
+    """
+
+    def __init__(self, response):
+        self.response = response
+
+    @property
+    def status(self):
+        return self.response.status
+
+    @property
+    def headers(self):
+        return self.response.headers
+
+    @property
+    def data(self):
+        return self.response.content
+
+
+class Request(transport.Request):
+    """Requests request adapter.
+
+    This class is used internally for making requests using asyncio transports
+    in a consistent way. If you use :class:`AuthorizedSession` you do not need
+    to construct or use this class directly.
+
+    This class can be useful if you want to manually refresh a
+    :class:`~google.auth.credentials.Credentials` instance::
+
+        import google.auth.transport.aiohttp_req
```

Comment:
I may have raised this before, but I am -1 on abbreviations. is there a motivation for naming this `aiohttp_req`? Instead of perhaps `aiohttp` or `aiohttp_request`?

Reply:
The motivation was the length, could change this to aiohttp_request though if that would be better. I'm worried about the name change to aiohttp just because of naming collision to the aiohttp library.

Reply:
Fixed, changed to aiohttp_requests in all 3 PRs